Redesigned Dashboard!

justinpotts opened the pull request:
Live preview: http://justinpotts.github.io/mozwebqa-dashboard/#/issues

Commit:
Redesign updates, including profile pictures, labels, issue counts, and collapsing.

davehunt commented on index.html:
```diff
@@ -10,8 +10,9 @@
-        <link rel="stylesheet" href="css/bootstrap.min.css">
-        <link rel="stylesheet" href="css/bootstrap-responsive.min.css">
+        <link rel="stylesheet" href="css/bootstrap.css">
+        <link rel="stylesheet" href="css/bootstrap.css">
```
Duplicate of previous line?

davehunt commented on index.html:
```diff
@@ -20,28 +21,50 @@
+              <ul class="nav nav-sidebar">
+                <li><a href="https://quality.mozilla.org/teams/web-qa/">About Web QA</a></li>
+                <li><a href="https://wiki.mozilla.org/QA/Execution/Web_Testing">Wiki</a></li>
+                <li><a href="https://oneanddone.mozilla.org/tasks/available/?search=&team=6">Contribute</a></li>
```
<strike>The ampersand should be URL escaped.</strike>

davehunt replied:
Ignore that, the GitHub highlighting was throwing me off.. this is fine in a query string 😄

davehunt commented on codebits.txt:
```diff
@@ -0,0 +1,85 @@
+Issues
```
Is this file used? I couldn't see any references to it...

justinpotts replied:
Heh, my bad on that. That was left over from development when I was storing some code in case I needed to revert changes. Deleting...

davehunt commented:
I think this is looking great! I have added a few comments, but with those addressed, I'm more than happy to merge it. I thought I'd point out the few differences in functionality from the current dashboard in case these are easy enough to include.

- The current dashboard shows a number next to each repository name, which indicates the total number of issues/pulls for that repository. I'm actually fine with dropping this as I don't feel like it adds value.
- The current dashboard allows each repository to be collapsed by clicking on the name. I quite like this feature, but it's not very clear in the current dashboard that it exists or when a repository is collapsed. Could we add this to the new dashboard with a little icon to indicate the state (or simply a text link of 'collapse/expand')?
- The current dashboard includes avatars for assignees. This is another feature that I don't feel adds value as it is. Perhaps if the avatar was larger it would be a more useful way to quickly identify familiar assignees. I'm personally happy with dropping this feature, perhaps to reintroduce it at a later date?
- The current dashboard shows the labels for each issue/pull. I know we discussed this, but thought it was worth highlighting here. This is something I'm happy to drop for now and reintroduce in a later enhancement.

Let's see if there's any other feedback from @mozilla/web-qa-sorcerers

bobsilverberg commented:
Looks good Justin! One thing that's missing that I don't think has been mentioned is that in the old version each issue had a grey bubble with a number next to it indicating the number of comments added to the issue. I think that's helpful to show you how much, if any, activity has happened on the issue, at a glance.

Commit:
Fix merge conflicts

justinpotts commented:
@davehunt @bobsilverberg: I've addressed all of your items and squashed my commits. New additions include labels for the amount of PRs and Issues, collapsing repos along with indicators to tell you whether it's open or not, avatars for those who are assigned, and labels for each issue.

I did not add photos or labels to pull requests, because if I'm not mistaken, we have never used any of these on our pull requests, and if we have, I don't think we use them often enough to warrant extra space / clutter in the UI.

Let me know what you guys think!

davehunt commented:
A few comments, and I'll indicate which should be resolved in this pull request, and which can be new issues (enhancements):

* We should have a little speech bubble (or similar) with a number in to indicate the number of *comments* for each issue/pull. This exists in the current dashboard and is a good indication of how active or contentious the issue/pull is. Please address this in this pull request.
* The larger avatars look great, but we should either remove the whitespace when there is nobody assigned, or add a placeholder avatar. My preference is the former. This can be a later enhancement.
* It's not a usual workflow for us, but I think we should support assignee avatars for pull requests.
* We used to include the color of the label from GitHub. I'm conflicted because I do like the grey labels that you've added, but the traffic-light system is useful for difficulty and priority. Perhaps there's a compromise, where the grey label could have a slice of colour. Something like:  <img width="275" alt="screenshot 2016-06-10 12 52 14" src="https://cloud.githubusercontent.com/assets/122800/15963468/347de5c4-2f0a-11e6-8c23-e2c005d1360f.png">

I'm also thinking that it shouldn't be too hard to make this project a generic dashboard for GitHub, which would be a valuable contribution to the community. Let's keep that in mind as we work on new features, making sure they're not specific to Mozilla.

Commit:
Minor UI tweaks from code review

justinpotts commented:
@davehunt, latest commit has added all of your suggestions! :)

Commit:
Add pointer cursor to PR repo titles too

Commit:
Update label borders and coloring
